samza-core/src/main/java/org/apache/samza/operators/impl/SendToTableOperatorImpl.java

```diff
@@ -54,7 +56,12 @@ protected void handleInit(Context context) {
   @Override
   protected CompletionStage<Collection<KV<K, V>>> handleMessageAsync(KV<K, V> message, MessageCollector collector,
       TaskCoordinator coordinator) {
-    return table.putAsync(message.getKey(), message.getValue(), sendToTableOpSpec.getArgs())
+    if (message.getValue() instanceof UpdateMessage) {
+      throw new SamzaException("Incorrect use of .sendTo operator with UpdateMessage value type. "
+          + "Please use the following method on MessageStream- sendTo(Table<KV<K, UpdateMessage<U, V>>> table,"
+          + "UpdateOptions updateOptions).");
```

Comment:
I would have definitely preferred to do it during construction over doing it `handleAsync`. Type erasure in java complicates creating an instance of a generic type parameter as type V is not known until runtime. I think it's possible to do it using some code like below if we had the `Class<V>` as parameter. 
```
public static class Container<E> {
    private Class<E> clazz;

    public Container(Class<E> clazz) {
        this.clazz = clazz;
    }

    public E createContents() throws Exception {
        return clazz.newInstance();
    }
}
```
Please let me know if know of any nitfy reflection hacks to get around this.

Reply:
looks like thats the case. Maybe we can skip it for now; ideally, we should have a this part of the `OperatorSpec` so that we can enable validations on these key and value types.